File: Clases/24_04_03-Actividad/24_04_03-Actividad_1/Views/MainForm.cs
using _24_04_03_Actividad_1.Models;
using _24_04_03_Actividad_1.Services;
using _24_04_03_Actividad_1.Views;
namespace _24_04_03_Actividad_1
{
    public partial class MainForm : Form
    {
        private User? _CurrentUser;
        private UserService _UserService = UserService.GetInstance();
        public MainForm(User currentUser)
        {
            InitializeComponent();
            this._CurrentUser = currentUser;

            this.LoadUsername();
            this.LoadProfilePic();
        }

        private void LoadUsername()
        {
            NameLbl.Text = this._CurrentUser.Name;
        }

        private void LoadProfilePic()
        {
            string appPath = Application.StartupPath;
            try
            {
                ProfilePic.Image = Image.FromFile(Path.Combine(appPath, "Uploads", this._CurrentUser.ProfilePic));
            }
            catch (FileNotFoundException)
            {
                ProfilePic.Image = Image.FromFile(Path.Combine(appPath, "Uploads/default.jpg"));
            }
        }

        private void UploadPicBtn_Click(object sender, EventArgs e)
        {
            if (ProfilePicLoader.ShowDialog() == DialogResult.OK)
            {
                Image picImage = Image.FromFile(ProfilePicLoader.FileName);

                // Guardar la imagen en una carpeta dentro de la aplicación
                string fileName = Path.GetFileName(ProfilePicLoader.FileName);
                string filePath = Path.Combine(Application.StartupPath, "Uploads", fileName);
                picImage.Save(filePath);

                ProfilePic.Image = Image.FromFile(ProfilePicLoader.FileName);
                _UserService.UpdateUserPic(this._CurrentUser.Username, fileName);
            }
        }

        private void calculadoraToolStripMenuItem1_Click(object sender, EventArgs e)
        {
            Form calculator = new CalculatorForm(this._CurrentUser);
            calculator.ShowDialog();
        }

        private void concatenadorToolStripMenuItem_Click(object sender, EventArgs e)
        {
            Form concatenator = new ConcatForm(this._CurrentUser);
            concatenator.ShowDialog();
        }
    }
}

File: Clases/24_04_09-Actividad/24_04_03-Actividad_1-VB/Views/MainForm.vb
Imports System.IO

Public Class MainForm

    Public Sub New()
        InitializeComponent()

        LoadUsername()
        LoadProfilePic()
    End Sub
    Private Sub LoadPicBtn_Click(sender As Object, e As EventArgs) Handles UploadPicBtn.Click
        Dim usersService As New UsersService()
        If ProfilePicLoader.ShowDialog() = DialogResult.OK Then
            Dim picImage As Image = Image.FromFile(ProfilePicLoader.FileName)

            Dim fileName As String = Path.GetFileName(ProfilePicLoader.FileName)
            Dim filePath As String = Path.Combine(Application.StartupPath, "Uploads", fileName)
            picImage.Save(filePath)

            ProfilePicBox.Image = Image.FromFile(ProfilePicLoader.FileName)
            usersService.UpdateUserPic(CurrentUser.Username, fileName)
        End If
    End Sub

    Private Sub LoadUsername()
        MessageBox.Show($"Hola {CurrentUser.FirstName}")
        CurrentUserLbl.Text = CurrentUser.FirstName
    End Sub

    Private Sub LoadProfilePic()
        Dim appPath As String = Application.StartupPath
        Try
            ProfilePicBox.Image = Image.FromFile(Path.Combine(appPath, "Uploads", CurrentUser.ProfilePicture))
        Catch ex As FileNotFoundException
            ProfilePicBox.Image = Image.FromFile(Path.Combine(appPath, "Uploads/default.jpg"))
        End Try
    End Sub

    Private Sub CalculadoraToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles CalculadoraToolStripMenuItem.Click
        Dim calculatorForm As New CalculatorForm()
        calculatorForm.ShowDialog()
    End Sub

    Private Sub ConcatenadorToolStripMenuItem_Click(sender As Object, e As EventArgs) Handles ConcatenadorToolStripMenuItem.Click
        Dim concatForm As New ConcatForm()
        concatForm.ShowDialog()
    End Sub

    Private Sub MainForm_FormClosed(sender As Object, e As FormClosedEventArgs) Handles MyBase.FormClosed
        ''Cierra el proceso principal (en .NET es el formulario de inicio(LoginForm))
        LoginForm.Close()
    End Sub
End Class